stats/src/main/java/io/airlift/stats/QuantileDigest.java

```diff
@@ -185,11 +186,67 @@ public void merge(QuantileDigest other)
     }
 
     /**
-     * Gets the values at the specified quantiles +/- maxError. The list of quantiles must be sorted
-     * in increasing order, and each value must be in the range [0, 1]
+     * Get a lower bound on the quantiles for the given proportions. A returned q quantile is guaranteed to be within
+     * the q - maxError and q quantiles.
+     *
+     * The input list of quantile proportions must be sorted in increasing order, and each value must be in the range [0, 1]
      */
-    public List<Long> getQuantiles(List<Double> quantiles)
+    public List<Long> getQuantilesLowerBound(List<Double> quantiles)
+    {
+        checkArgument(Ordering.natural().isOrdered(quantiles), "quantiles must be sorted in increasing order");
+        for (double quantile : quantiles) {
+            checkArgument(quantile >= 0 && quantile <= 1, "quantile must be between [0,1]");
+        }
+
+        ArrayList<Double> reversedQ = new ArrayList(quantiles);
```

Comment:
Nit: don't abbreviate variables (so name this `reversedQuantiles` or perhaps just `reversed`)